File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        c=[]
        l=0
        mx=0
        for i in range(len(s)):
            while s[i] in c:
                c.remove(s[l])
                l+=1
            c.append(s[i])
            mx=max(mx,i-l+1)
        return mx

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int>length(256,-1);
        int left=0;
        int right=0;
        int ans=0;
        while(right<s.size()){
            if(length[s[right]] != -1){
                left=max(left,length[s[right]]+1);
            }
            length[s[right]]=right;
            ans=max(ans,right-left+1);
            right++;
        }
        return ans;
    }
};


